### DISCUSSION TITLE
Pocketbase instance for multiple users

### DISCUSSION THREAD

**Discussion by mrbu-solutions** [pocketbase/pocketbase#4924] [Q&A]
What is the recommended way of handling multiple users using pocketbase authentication?

I'm using sveltekit and most of the tutorials I've watched had put the pocketbase instance in sveltekit event.locals (i.e. event.locals.pb). This means that every request on the server, a new pocketbase instance will get instantiated which I think is expensive.

Is this the best practice, or should I put pocketbase instance in a singleton class instead and not pass to event.locals?

For context, I'm creating an app that can support multiple databases, and one of them is pocketbase.

**Comment by ganigeorgiev** [comment]
> ... a new pocketbase instance will get instantiated which I think is expensive.

You are overthinking it. Creating a new SDK instance is not much different than calling `const a = {}` (HTTP requests are sent only when the related SDK methods are invoked).

> Is this the best practice, or should I put pocketbase instance in a singleton class instead and not pass to event.locals?

If you use a global object you are introducing a security vulnerability (this is not unique to PocketBase) because requests from different users will write over the same object (in your case `pb.authStore`).

You need a new local instance per request because you need to store somewhere the unique request auth data.

_As a side-note, I'll paste a reply from another discussion:_

> I've posted this several times but I don't recommend using a JS SSR framework with PocketBase as most likely you'll mess up the security, especially if you are new to web development.
>
> Alternatively, you may not actually need PocketBase because if you are using SvelteKit with the node adapter you already have a backend, aka. the node server, and if you need just a simple database you can use SQLite directly with Drizzle, Prisma, etc. For auth there are also well documented and maintained libraries like [Lucia](https://github.com/lucia-auth/lucia).
>
> PocketBase is not just a database, it is a backend. The easiest way to use it is by interacting directly with your PocketBase server from the client-side (eg. with a SPA). Some more ramblings on this you can read in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032.

**Reply by mrbu-solutions** [reply]
Thank you very much.
